README.md

```diff
@@ -2,45 +2,162 @@
 Bot for telegram channel that 'll publish news based on RSS feed : 
 https://t.me/SwissITNews
 
-# How to use this bot :
+# How to use this bot
 
-1) Download or fork the content of this repository
+### Quick Start with Docker
 
-2) Get a Raspberry Pi and install the needed libraries : 
+1. Clone or Download the content of this repository.
 
-```bash
-virtualenv .venv -p python3
-source .venv/bin/activate
-pip install -r requirements.txt
-```
+2. Ensure Docker is installed on your system.
 
-3) Create your telegram channel , for example here is mine : 
-https://t.me/SwissITNews
+3. Configuration options:
 
-4) Create a bot by sending a message to "@BotFather" on Telegram : https://telegram.me/BotFather
+   You have two options for configuring the bot: using a `.env` file or a TOML file named `configuration.toml`. You can use one of them or both, but remember that environment variables (if set) will take precedence over the settings in the TOML file.
 
-![Bot Father](https://cdn1.telesco.pe/file/NZabVMBzG_0Rntl0ZsfxwQHwpbwdyznyiF-TLzQu7vdYlr8FcuQPbAxuhZRGYjkQUL-WkfdrmqDWiRsVaiM_CVC_Ugb6_2EYdPITtANtCTqXusUUTyXQzzGA9hYaZuahAwYm27-aaBgoh-zs3lXO3UfluRrQNeO0CiijcpNtEJI3ug1aVjyOICWdLsz4awsD92acA8VctWGyF7ZWRgQnpGdbfDkChucbh9BnPfTlxB8LeAVZmAV7YqF6eG910usqlImOPdRZGIJq2Ky42hc55ZbELry3t_bx4Rxbev8DQLM9iQhoKzmAjZ2m02DEgaCPKMIZ4X-hbOj2X87zgyTWmA.jpg)
+   - **Using a `.env` file:**
 
-5) Add your bot to your channel and setup permissions to send messages
+     Create a `.env` file at the root of the project with the following content, replacing the placeholders with your actual data:
 
-6) Replace your bot token , channel ID and RSS feeds in the script 
+     ```
+     # .env file
+     TIME_INTERVAL_MIN=1
+     ENTRY_MAX_TIME_OLD=1200
+     BOT_TOKEN="YOUR BOT TOKEN"
+     CHANNEL_ID="@YourChannel"
+     RSS_URLS='https://korben.info/feed;https://www.ictjournal.ch/taxonomy/term/404/feed;https://www.ictjournal.ch/taxonomy/term/31/feed;https://www.ictjournal.ch/taxonomy/term/406/feed;https://www.newsd.admin.ch/newsd/feeds/rss?lang=fr&org-nr=1&offer-nr=308;https://feeds.feedburner.com/TheHackersNews;https://redmondmag.com/rss-feeds/columns.aspx;https://redmondmag.com/rss-feeds/in-depth.aspx;https://redmondmag.com/rss-feeds/news.aspx;https://redmondmag.com/rss-feeds/tech-library.aspx;https://www.xavierstuder.com/feed;https://www.thelazyadministrator.com/feed;https://4sysops.com/feed;https://www.thomasmaurer.ch/feed;https://www.cyberciti.biz/atom/atom.xml;https://blog.codinghorror.com/rss;https://www.schneier.com/feed/atom'
```

Comment:
I would recommend the following synthax here to keep the URL readable in the `.env` file.

`RSS_URLS="https://korben.info/feed
https://www.ictjournal.ch/taxonomy/term/404/feed
https://www.ictjournal.ch/taxonomy/term/31/feed
https://www.ictjournal.ch/taxonomy/term/406/feed
https://www.newsd.admin.ch/newsd/feeds/rss?lang=fr&org-nr=1&offer-nr=308
https://feeds.feedburner.com/TheHackersNews
https://mspoweruser.com/feed/
https://www.numerama.com/feed/"`

Additional gain : no need of a particular separator in` str.split()`.

Reply:
I ran a few tests with your suggestion and it doesn't work as expected on my side. It only takes the first URL and the single quote at the beginning is part of the string. 
Even when I put everything on one line, if I keep the single quote at the beginning the single quote is passed to the whole string and messes things up. 

So I've updated my .env file as follow and I have better results so far : 


```
# Development settings
TIME_INTERVAL_MIN=20
ENTRY_MAX_TIME_OLD=1200
BOT_TOKEN=1234567:MYBOTTOKEN
CHANNEL_ID=@MyTelegramChannel
RSS_URLS=https://korben.info/feed,https://www.ictjournal.ch/taxonomy/term/404/feed,https://www.ictjournal.ch/taxonomy/term/31/feed,https://www.ictjournal.ch/taxonomy/term/406/feed,https://www.newsd.admin.ch/newsd/feeds/rss?lang=fr&org-nr=1&offer-nr=308,https://feeds.feedburner.com/TheHackersNews,https://redmondmag.com/rss-feeds/columns.aspx,https://redmondmag.com/rss-feeds/in-depth.aspx,https://redmondmag.com/rss-feeds/news.aspx,https://redmondmag.com/rss-feeds/tech-library.aspx,https://www.xavierstuder.com/feed,https://www.thelazyadministrator.com/feed,https://4sysops.com/feed,https://www.thomasmaurer.ch/feed,https://www.cyberciti.biz/atom/atom.xml,https://blog.codinghorror.com/rss,https://www.schneier.com/feed/atom

# Docker container settings
TZ=Europe/Zurich
```

I also had to adapt the `split()` to `split(',')`

I will let it run a few ours and check if it works correctly. At the moment I had 1 success and will monitor for a few more hours. If it works correctly like that then I will update the README.md  and the `split(',')` in the python bot. 

_fyi , I'm passing the .env file when building the docker container. Idk if it could be linked to the issue I am having with the quotes in the_ `.env` _file._
`docker run -d --name swiss-it-tech-news-bot --env-file /mypath/tomyenv/.env swiss-it-tech-news`

Reply:
If I remember well I also did try with the double quotes and it didn't work. 

I've just tried again and I'm sure I already did this test because I had the same issue in the logs where only the first URL is passed in the string with the double quote contained in the string. See below screenshot  : 
![2024-03-01-16_57_56-swiss-it-tech-news](https://github.com/AlexIn-Tech/Swiss-IT-Tech-News/assets/18593496/726d8eb4-d109-4813-9c2e-b66e89dabf91)

Let's wait and see but I'm pretty sure it will not work.

Reply:
I can confirm (based on the logs) it only checks for the first URL and the string of the first URL is not correct because it contains the double quote inside the string. 
![2024-03-01-17_17_17-swiss-it-tech-news](https://github.com/AlexIn-Tech/Swiss-IT-Tech-News/assets/18593496/a8f9270e-99e1-41ce-85bb-63389206926b)

Reply:
I've copied and pasted it and this is the exact same issue. 
I even tried with Docker Desktop just to have a look if it could be linked to my server but no I have the exact same issue : 
![2024-03-01 23_16_16-swiss-it-tech-news-bot - Container - Docker Desktop  DEV](https://github.com/AlexIn-Tech/Swiss-IT-Tech-News/assets/18593496/cc8b73bf-aebd-4fb6-aa7f-f8d0ae0c0808)

It only works with this formatting : 
```
# Development settings
TIME_INTERVAL_MIN=20
ENTRY_MAX_TIME_OLD=1200
BOT_TOKEN=1234567:MYBOTTOKEN
CHANNEL_ID=@MyTelegramChannel
RSS_URLS=https://korben.info/feed,https://www.ictjournal.ch/taxonomy/term/404/feed,https://www.ictjournal.ch/taxonomy/term/31/feed,https://www.ictjournal.ch/taxonomy/term/406/feed,https://www.newsd.admin.ch/newsd/feeds/rss?lang=fr&org-nr=1&offer-nr=308,https://feeds.feedburner.com/TheHackersNews,https://redmondmag.com/rss-feeds/columns.aspx,https://redmondmag.com/rss-feeds/in-depth.aspx,https://redmondmag.com/rss-feeds/news.aspx,https://redmondmag.com/rss-feeds/tech-library.aspx,https://www.xavierstuder.com/feed,https://www.thelazyadministrator.com/feed,https://4sysops.com/feed,https://www.thomasmaurer.ch/feed,https://www.cyberciti.biz/atom/atom.xml,https://blog.codinghorror.com/rss,https://www.schneier.com/feed/atom

# Docker container settings
TZ=Europe/Zurich
```

Reply:
I tried as you suggested with a copy paste of this exact content for the ` .env ` file and I am having the same issue : 
![2024-03-01 23_16_16-swiss-it-tech-news-bot - Container - Docker Desktop  DEV](https://github.com/AlexIn-Tech/Swiss-IT-Tech-News/assets/18593496/491bf33f-169d-4f7d-a8b5-fb76293da08f)

I even tried with Docker Desktop in case an issue could come frome my server but it's the same result. 

The only formatting that works for this `.env` file is this one : 
```
# Development settings
TIME_INTERVAL_MIN=20
ENTRY_MAX_TIME_OLD=1200
BOT_TOKEN=1234567:MYBOTTOKEN
CHANNEL_ID=@MyTelegramChannel
RSS_URLS=https://korben.info/feed,https://www.ictjournal.ch/taxonomy/term/404/feed,https://www.ictjournal.ch/taxonomy/term/31/feed,https://www.ictjournal.ch/taxonomy/term/406/feed,https://www.newsd.admin.ch/newsd/feeds/rss?lang=fr&org-nr=1&offer-nr=308,https://feeds.feedburner.com/TheHackersNews,https://redmondmag.com/rss-feeds/columns.aspx,https://redmondmag.com/rss-feeds/in-depth.aspx,https://redmondmag.com/rss-feeds/news.aspx,https://redmondmag.com/rss-feeds/tech-library.aspx,https://www.xavierstuder.com/feed,https://www.thelazyadministrator.com/feed,https://4sysops.com/feed,https://www.thomasmaurer.ch/feed,https://www.cyberciti.biz/atom/atom.xml,https://blog.codinghorror.com/rss,https://www.schneier.com/feed/atom

# Docker container settings
TZ=Europe/Zurich
```